src/solve/mirk.jl

```diff
@@ -42,10 +42,10 @@ function SciMLBase.__init(prob::BVProblem, alg::AbstractMIRK; dt = 0.0,
     mesh_dt = diff(mesh)
 
     chunksize = pickchunksize(N * (Nig - 1))
-    __alloc = @closure x -> __maybe_allocate_diffcache(vec(x), chunksize, alg.jac_alg)
+    __alloc = @closure x -> __maybe_allocate_diffcache(vec(__similar(x)), chunksize, alg.jac_alg)
 
-    fᵢ_cache = __alloc(similar(X))
-    fᵢ₂_cache = vec(similar(X))
+    fᵢ_cache = __alloc(__similar(X))
+    fᵢ₂_cache = vec(__similar(X))
```

Comment:
```suggestion
    fᵢ_cache = __alloc(zero(X))
    fᵢ₂_cache = vec(zero(X))
```

Why not just use `zero` everywhere?

Reply:
Yeah, changed to `zero` everywhere